internal/controllers/port/actuator.go

```diff
@@ -263,3 +281,22 @@ func newActuator(ctx context.Context, controller interfaces.ResourceController,
 		k8sClient: controller.GetK8sClient(),
 	}, nil, nil
 }
+
+// portSecurityEnabled checks if port security is enabled based on the given state.
+func portSecurityEnabled(portSecurityState orcv1alpha1.PortSecurityState, networkStatus orcv1alpha1.NetworkStatus) bool {
+	switch portSecurityState {
+	case orcv1alpha1.PortSecurityEnabled:
+		return true
+	case orcv1alpha1.PortSecurityInherit:
+		// PortSecurity at the network level is enabled by default
+		// https://docs.openstack.org/api-ref/network/v2/#port-security
+		if networkStatus.Resource == nil || networkStatus.Resource.PortSecurityEnabled == nil {
+			return true
+		}
+		return *networkStatus.Resource.PortSecurityEnabled
```

Comment:
Is there any reason we're setting this explicitly here rather than just not setting the value and letting OpenStack do it?

Reply:
Do you mean whether we need the Inherit PortSecurityState ?

Reply:
I mean: when the value is `PortSecurityInherit` we should probably just not set PortSecurity in the OpenStack API request and let OpenStack handle it. Here we're explicitly setting it to the value from the network.